src/relax/op/distributed/distributed.cc

```diff
@@ -85,5 +86,84 @@ TVM_REGISTER_OP("relax.dist.redistribute")
     .set_attr<FInferStructInfo>("dist.FInferStructInfo", InferDistStructInfoRedistribute)
     .set_attr<Bool>("FPurity", Bool(true));
 
+StructInfo InferStructInfoRtoS(const Call& call, const BlockBuilder& ctx) {
+  TensorStructInfo input_sinfo = GetUnaryInputTensorStructInfo(call, ctx);
+  DataType output_dtype = input_sinfo->dtype;
+
+  const auto* attrs = call->attrs.as<ScatterCollectiveAttrs>();
+  int num_workers = attrs->num_workers;
+
+  arith::Analyzer* analyzer = ctx->GetAnalyzer();
+  auto input_shape = input_sinfo->GetShape();
+  CHECK(input_shape.defined()) << "input tensor of scatter_from_worker0 should have defined shape.";
+
+  if (analyzer->CanProve(floormod(input_shape.value()[0], PrimExpr(num_workers))) != 0) {
```

Comment:
`input_shape.value()[0]` should be `input_shape.value()[attrs->axis]`.  As currently written, we check the divisibility of the outermost axis instead of the axis being split.